File: Java/Java source-code/evaluateWeightedProduct.java


public class evaluateWeightedProduct {
    public static double evaluateWeightedProduct(final Double[] values, final Double[] weights, final Integer begin,
                                                 final Integer length) {
        double product = Double.NaN;
        product = 1.0;
        for (int i = begin; i < begin + length; i++) {
            product *= Math.pow(values[i], weights[i]);
        }
        return product;
    }
}

File: Methods/Methods-C++/evaluateWeightedProduct.cpp
#include <iostream>

double evaluateWeightedProduct( double values[],  double weights[],  int begin, int lengths, int n) {
    double product = 1.0;
     
    for (int i = begin; i < begin + lengths; i++) {
        product *= pow(values[i], weights[i]);
    }
    return product;
}